pkgs/by-name/ks/ksoloti/package.nix

```diff
@@ -0,0 +1,181 @@
+        sha256 = "sha256-bFAsgWAC+Q1PdgUKZCnDp+DYQgQiLL/y3Old13O6aEA=";
+      };
+
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ systemd ];
+
+    patches = [
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/ksoloti/ksoloti/${ksolotiVersion}/platform_linux/src/libusb.stdfu.patch";
+        sha256 = "sha256-bYzcE20fnwBTZUZmYYHgkW7SekO6pgoj6AaTGIw8kqQ=";
+      })
+    ];
+  };
+
+  patched-dfu-util = dfu-util.overrideAttrs (oldAttrs: {
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ libusb ];
+  });
+
+  chibios = stdenv.mkDerivation {
+    name = "chibios";
+    version = chibiOSVersion;
+    src = fetchFromGitHub {
+      owner = "ChibiOS";
+      repo = "ChibiOS";
+      rev = "ver${chibiOSVersion}";
+      hash = "sha256-ILb6OluhriztSRbVIqb4NGzmbSlH9cb+NOw3b58Hz0I=";
+    };
+
+    buildInputs = [ unzip ];
+
+    patchPhase = ''
+      cd ext
+      unzip -o fatfs-0.9-patched.zip
+      cd ..
+    '';
+
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir -p $out
+      cp -r * $out
+    '';
+  };
+
+  ksoloti-runtime = stdenv.mkDerivation {
+    name = "ksoloti-runtime";
+    inherit src;
+    buildInputs = [ chibios ];
+    nativeBuildInputs = [ patched-dfu-util gcc-arm-embedded-6 ];
+
+    patchPhase = ''
+      cp -r ${chibios} ./chibios
+    '';
+
+    buildPhase = ''
+      export axoloti_release=$(pwd)
+
+      for dir in firmware firmware_axoloti_legacy; do
+        cd $axoloti_release/$dir
+        mkdir -p build/obj
+        mkdir -p build/lst
+        make
+
+        cd $axoloti_release/$dir/flasher
+        mkdir -p flasher_build/lst
+        mkdir -p flasher_build/obj
+        make
+
+        cd $axoloti_release/$dir/mounter
+        mkdir -p mounter_build/lst
+        mkdir -p mounter_build/obj
+        make
+      done
+
+      cd $axoloti_release
+    '';
+
+    installPhase = ''
+      mkdir -p $out
+      cp -r firmware $out
+      cp -r firmware_axoloti_legacy $out
+      cp -r chibios $out
+    '';
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "ksoloti";
+  version = ksolotiVersion;
+
+  inherit src;
+
+  buildInputs = [ ksoloti-runtime zulu ant ];
+  nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+
+  buildPhase = "ant -Dbuild.version=${version} -Dshort.version=${builtins.head (builtins.split "-" version)}";
+
+  installPhase = ''
+    chmod +x Ksoloti.sh
+    mkdir -p $out/opt/ksoloti
+    cp -r * $out/opt/ksoloti
+    cp -r ${ksoloti-runtime}/* $out/opt/ksoloti
+
+    wrapProgram "$out/opt/ksoloti/Ksoloti.sh" \
+      --prefix PATH : '${lib.makeBinPath [ zulu gcc-arm-embedded-6 ]}' \
+      --prefix LD_LIBRARY_PATH : '${lib.makeLibraryPath [ gcc-arm-embedded-6 ]}' \
+      --set JAVA_HOME '${zulu}' \
+      --set axoloti_release $out/opt/ksoloti \
+      --set axoloti_home AXOLOTI_HOME
+
+    substituteInPlace $out/opt/ksoloti/Ksoloti.sh --replace "'AXOLOTI_HOME'" '"$XDG_DATA_HOME/ksoloti"'
```

Comment:
this is a bit of a hack, I guess. If you know a better way to add the the variable without it being single quoted, let me know! :pray:

Reply:
I added a fallback. :+1: